telemetry/config.go

```diff
@@ -0,0 +1,233 @@
+package telemetry
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"log/slog"
+	"os"
+	"time"
+
+	"github.com/hashicorp/go-metrics"
+	"go.opentelemetry.io/contrib/bridges/otelslog"
+	"go.opentelemetry.io/contrib/instrumentation/host"
+	"go.opentelemetry.io/contrib/instrumentation/runtime"
+	"go.opentelemetry.io/contrib/otelconf/v0.3.0"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/exporters/stdout/stdoutlog"
+	"go.opentelemetry.io/otel/exporters/stdout/stdoutmetric"
+	"go.opentelemetry.io/otel/exporters/stdout/stdouttrace"
+	logglobal "go.opentelemetry.io/otel/log/global"
+	logsdk "go.opentelemetry.io/otel/sdk/log"
+	metricsdk "go.opentelemetry.io/otel/sdk/metric"
+	tracesdk "go.opentelemetry.io/otel/sdk/trace"
+	"go.yaml.in/yaml/v3"
+)
+
+var sdk otelconf.SDK
+var shutdownFuncs []func(context.Context) error
+
+func init() {
+	err := doInit()
+	if err != nil {
+		panic(err)
+	}
+}
+
+func doInit() error {
```

Comment:
quite a lot going on here - do you see this simplifying in the future?

Reply:
Upstream, I think otelconf will add the file exporters so most of the cosmos extra will go away although we probably want to keep the runtime instrumentation stuff. And we may also want some other cosmos extra config options